File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        // 1. 이름과 그리움 점수를 매핑 (Map 사용)
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < name.length; i++) {
            map.put(name[i], yearning[i]);
        }

        int[] result = new int[photo.length];
        
        // 2. 사진 순회
        for (int i = 0; i < photo.length; i++) {
            int score = 0;
            // 3. 사진 속 인물의 점수 합산
            for (String person : photo[i]) {
                score += map.getOrDefault(person, 0); // 없는 인물은 0점
            }
            result[i] = score;
        }
        
        return result;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    // 1. 이름과 그리움 점수를 매핑 (Object 또는 Map)
    const scoreMap = {};
    name.forEach((n, i) => {
        scoreMap[n] = yearning[i];
    });

    // 2. 사진 배열 순회하며 점수 계산
    return photo.map(p => 
        p.reduce((acc, cur) => acc + (scoreMap[cur] || 0), 0)
    );
}
